src/app/projects/add-edit-project/add-edit-project.component.ts

```diff
@@ -10,7 +10,7 @@ import { StorageService } from 'app/services/storage.service';
 import { ProjectService } from 'app/services/project.service';
 import { DocumentService } from 'app/services/document.service';
 import { CkUploadAdapter } from 'app/shared/utils/ck-upload-adapter';
-import { Project, ProjectType } from 'app/models/project';
+import { Project, ProjectShapefile, ProjectType, ProjectShapefileOrDocument } from 'app/models/project';
```

Comment:
Not crucial at all, but I believe you can combine types like this. 
type Combined = Partial<TypeA & TypeB>;

Reply:
After further inspection, it looks like the only property on that type is ID. I thought you were trying to merge two types into one, with all of the common properties.

Reply:
I appreciate the feedback anyway! I wasn't wild about this solution but it works